pkg/apply/event/event.go

```diff
@@ -64,9 +67,12 @@ type ResourceAction int
 const (
 	ApplyAction ResourceAction = iota
 	PruneAction
+	DeleteAction
+	WaitAction
 )
 
-type ResourceGroup struct {
+type ActionGroup struct {
+	Name        string
```

Comment:
What is the intent of adding the `Name` field. Based on the code from other file, it is assigned to be the TaskName. This is kind of duplicate with the values for `Action`.

Reply:
So the idea is that we want some way to distinguish different tasks of the same type. The most relevant example for the current functionality is when we create a separate Apply task for CRDs, but eventually we want to support additional ordering features which could lead to multiple tasks of the same type.
In this PR, the tasks just get named with the name of the type and then a number. So if we have two apply tasks, they will be named `apply-1` and `apply-2`. I have been looking at whether this is a better way to name them that reflects their purpose, but I haven't found a good solution yet.